main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1473,7 +1473,11 @@ private void updateFavPointBox() {
             // Favorite counts
             if (cache.getFavoritePoints() > 0) {
                 favoriteLine.left.setVisibility(View.VISIBLE);
-                favoriteLine.right.setText(cache.getFavoritePoints() + "×");
+                final StringBuilder favoriteString = new StringBuilder(String.valueOf(cache.getFavoritePoints()));
+                if (cache.getFavoritePoints() > 0 && cache.getFindsCount() > 0) {
+                    favoriteString.append(" (" + Math.round((float) cache.getFavoritePoints() / cache.getFindsCount() * 100) + "%)");
```

Comment:
Using concatenation inside append() is an anti pattern. Wondering that checkstyle didn't trigger here. Change to append().append().append() please.

If you reorder the math as ((float)(favorites*100))/finds, then you do one operation as (faster) integer instead of everything as float.

Also you are repeatedly calling the same accessor methods. You should always store to a temporary local variable, if you need the same method call value more than once (since you never know what is behind that call).

Reply:
I still cannot get used to this super ugly append().append() construction even though I know about immutable strings in Java :(

Done, thanks for comments.

Reply:
In plain Java, if you have just _one_ statement containing String concatenation, it doesn't even make a difference, since all current compilers replace such String concatenation by StringBuilder.append() calls during compilation. :)